### PR TITLE
websocket enabled check

### PR DESCRIPTION
**Pull request by pan-wang**
check whether websocket is enabled in configuration. So that we will not forward the upgrade header if websocket is not enabled. Also add Release cal to avoid memory leak

### CODE REVIEW

File: src/CommonLib/aspnetcoreconfig.h [hunk 1]
```diff
@@ -168,11 +169,17 @@ class ASPNETCORE_CONFIG : IHttpStoredContext
...
 
     BOOL
-        QueryStdoutLogEnabled()
+    QueryStdoutLogEnabled()
     {
         return m_fStdoutLogEnabled;
     }
 
+    BOOL
+    QueryWebSocketEnable()
```
Comment (bangbingsyb) [inline src/CommonLib/aspnetcoreconfig.h]: QueryWebSocketEnable**d** to keep name consistency?

File: src/RequestHandler/outofprocess/forwardinghandler.cpp [hunk 2]
```diff
@@ -165,7 +165,7 @@ FORWARDING_HANDLER::OnExecuteRequestHandler()
     //
     // Mark request as websocket if upgrade header is present.
     //
-    if (g_fWebSocketSupported)
+    if (pApplication->QueryConfig()->QueryWebSocketEnable())
```
Comment (jkotalik) [inline src/RequestHandler/outofprocess/forwardinghandler.cpp]: I think we should still check win8+ too?

Comment (jkotalik) [reply src/RequestHandler/outofprocess/forwardinghandler.cpp]: Remove g_fWebSocketSupported.

File: src/CommonLib/aspnetcoreconfig.cxx [hunk 3]
```diff
@@ -161,6 +162,7 @@ ASPNETCORE_CONFIG::Populate(
     BSTR                            bstrBasicAuthSection = NULL;
     BSTR                            bstrAnonymousAuthSection = NULL;
     BSTR                            bstrAspNetCoreSection = NULL;
+    BSTR                            bstrWebscoketSection = NULL;
```
Comment (nil4) [inline src/CommonLib/aspnetcoreconfig.cxx]: Typo: `Webscoket`

File: src/RequestHandler/outofprocess/forwardinghandler.cpp (hunk 2 above)
Comment (pan-wang) [reply src/RequestHandler/outofprocess/forwardinghandler.cpp]: no need to check win8 as on the websocket configuration section is empty win7 and QueryWebSocketEnable return false. g_fWebSocketedSupported is used for initialize winhttpwebsocket function pointers and static init/cleanup

File: src/CommonLib/aspnetcoreconfig.h (hunk 1 above)
Comment (pan-wang) [reply src/CommonLib/aspnetcoreconfig.h]: good point

File: src/CommonLib/aspnetcoreconfig.cxx (hunk 3 above)
Comment (pan-wang) [reply src/CommonLib/aspnetcoreconfig.cxx]: fixed

File: src/RequestHandler/outofprocess/forwardinghandler.cpp (hunk 2 above)
Comment (jkotalik) [reply src/RequestHandler/outofprocess/forwardinghandler.cpp]: Then shouldn't g_fWebSocketedSupported be equal to pApplication->QueryConfig()->QueryWebSocketEnabled()?

**Pull request merged**